File: app/src/main/java/com/botijasoftware/utils/shaders/Shader.java
package com.botijasoftware.utils.shaders;

import com.botijasoftware.utils.ResourceManager;
import com.botijasoftware.utils.ShaderProgram;

public class Shader {
	protected ShaderProgram program;
	protected String vsSource;
	protected String fsSource;
	protected int programid;

	public void LoadContent(ResourceManager resources) {
		
	}
	
	public void enable() {
		
	}

    public void reload(ResourceManager resources) {
        LoadContent(resources);
    }

	
}

File: app/src/main/java/com/botijasoftware/utils/shaders/Shader.kt
package com.botijasoftware.utils.shaders

import com.botijasoftware.utils.ResourceManager
import com.botijasoftware.utils.ShaderProgram

open class Shader {
    protected lateinit var program: ShaderProgram
    protected lateinit var vsSource: String
    protected lateinit var fsSource: String
    protected var programid: Int = 0

    fun LoadContent(resources: ResourceManager) {

    }

    open fun enable() {

    }

    fun reload(resources: ResourceManager) {
        LoadContent(resources)
    }


}
